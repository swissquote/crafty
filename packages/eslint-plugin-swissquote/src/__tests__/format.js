/* global describe, it, expect */

const { prepareCLIEngine, lint } = require("../../test_utils");
const engine = prepareCLIEngine("format");

describe("ES6 Formatting", () => {
  it("Gives no warning on correct code", () => {
    const result = lint(
      engine,
      `
module.exports = function initJS(gulp, config, watchers) {
  const js = config.js,
    jsTasks = [];

  for (const name in js) {
    if (!js.hasOwnProperty(name)) {
      continue;
    }

    const taskName = \`js_\${name}\`;

    if (!compileWithWebpack(js[name])) {
      gulp.task(taskName, jsTaskES5(gulp, config, watchers, js[name]));
      watchers.add(js[name].watch || js[name].source, taskName);
    }

    jsTasks.push(taskName);
  }

  gulp.task("js", jsTasks);

  return ["js"];
};
`
    );

    expect(result.messages).toMatchSnapshot();
    expect(result.warningCount).toBe(0);
    expect(result.errorCount).toBe(0);
  });

  it("Fails on badly formatted code", () => {
    const result = lint(
      engine,
      `
module.exports = function initJS(gulp, config, watchers) {
  const js = config.js,
     jsTasks = [];

  for (const name in js) {
    if (!js.hasOwnProperty(name)) {
      continue;
      }

    const taskName = \`js_\${name}\`;

    if (!compileWithWebpack(js[name])) {
      gulp.task(taskName, jsTaskES5(gulp, config, watchers, js[name]));
        watchers.add(js[name].watch || js[name].source, taskName);
    }

    jsTasks.push( taskName );
  }

  gulp.task("js", jsTasks);

  return ["js"];
};
`
    );

    expect(result.messages).toMatchSnapshot();
    expect(result.warningCount).toBe(0);
    expect(result.errorCount).toBe(4);
  });
});
