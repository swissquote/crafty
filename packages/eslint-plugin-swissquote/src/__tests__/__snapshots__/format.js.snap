// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ES6 Formatting Fails on badly formatted code 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        87,
        88,
      ],
      "text": "",
    },
    "line": 3,
    "message": "Delete \`·\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 8,
    "fix": Object {
      "range": Array [
        182,
        184,
      ],
      "text": "",
    },
    "line": 8,
    "message": "Delete \`··\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 14,
    "fix": Object {
      "range": Array [
        340,
        348,
      ],
      "text": "      ",
    },
    "line": 14,
    "message": "Replace \`········\` with \`······\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 18,
    "endColumn": 28,
    "endLine": 17,
    "fix": Object {
      "range": Array [
        431,
        441,
      ],
      "text": "taskName",
    },
    "line": 17,
    "message": "Replace \`·taskName·\` with \`taskName\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
]
`;

exports[`ES6 Formatting Gives no warning on correct code 1`] = `Array []`;
